contrib/grpc-contrib/src/main/java/com/salesforce/grpc/contrib/interceptor/DebugClientInterceptor.java

```diff
@@ -0,0 +1,119 @@
+/*
+ *  Copyright (c) 2019, Salesforce.com, Inc.
+ *  All rights reserved.
+ *  Licensed under the BSD 3-Clause license.
+ *  For full license text, see LICENSE.txt file in the repo root  or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+package com.salesforce.grpc.contrib.interceptor;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.ForwardingClientCall;
+import io.grpc.ForwardingClientCallListener;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.grpc.Status;
+
+/**
+ * {@code DebugClientInterceptor} intercepts the both outbound requests and inbound response to
+ * log the method name,status,headers and message as per given logging level.
+ *
+ */
+public class DebugClientInterceptor implements ClientInterceptor {
+    private final Logger logger = LoggerFactory.getLogger(DebugClientInterceptor.class);
+    private static final String REQUEST = "Request";
+    private static final String RESPONSE = "Response";
+
+    public enum Level {
+        NONE, STATUS, HEADERS, MESSAGE
+    }
+
+    private Level level = Level.NONE;
+
+    public DebugClientInterceptor(Level level) {
```

Comment:
It would be great if the user could specify a list of events to log using variadic parameters instead of only choosing one.

Reply:
Updated the constructor to have varargs